x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/format_selector.tsx

```diff
@@ -0,0 +1,131 @@
+      defaultMessage: 'Bytes (1024)',
+    }),
+  },
+};
+
+type FormatSelectorProps = Pick<IndexPatternDimensionPanelProps, 'data'> & {
+  selectedColumn: IndexPatternColumn;
+  currentIndexPattern: IndexPattern;
+  onChange: (newFormat?: { id: string; params?: Record<string, unknown> }) => void;
+};
+
+export function FormatSelector(props: FormatSelectorProps) {
+  const { selectedColumn, onChange } = props;
+
+  const currentFormat =
+    selectedColumn.params && 'format' in selectedColumn.params
+      ? selectedColumn.params.format
+      : undefined;
+
+  const selectedFormat = currentFormat?.id ? supportedFormats[currentFormat.id] : undefined;
+
+  const defaultOption = {
+    value: '',
+    label: i18n.translate('xpack.lens.indexPattern.defaultFormatLabel', {
+      defaultMessage: 'Default',
+    }),
+  };
+
+  const currentDecimals = currentFormat?.params?.decimals;
+
+  return (
+    <>
+      <EuiFormRow
+        label={i18n.translate('xpack.lens.indexPattern.columnFormatLabel', {
+          defaultMessage: 'Display format',
+        })}
+        display="rowCompressed"
+      >
+        <EuiComboBox
+          fullWidth
+          compressed
+          isClearable={false}
+          data-test-subj="indexPattern-dimension-format"
+          singleSelection={{ asPlainText: true }}
+          options={[defaultOption].concat(
+            Object.entries(supportedFormats).map(([id, format]) => ({
+              value: id || '',
+              label: format.title || id || '',
+            }))
+          )}
+          selectedOptions={
+            currentFormat
+              ? [
+                  {
+                    value: currentFormat.id || '',
+                    label: selectedFormat?.title || currentFormat.id || '',
+                  },
+                ]
+              : [defaultOption]
+          }
+          onChange={choices => {
+            if (choices.length === 0) {
+              return;
+            }
+
+            if (!choices[0].value) {
+              onChange();
+              return;
+            }
+            onChange({
+              id: choices[0].value,
+              params: { decimals: typeof currentDecimals === 'number' ? currentDecimals : 3 },
+            });
+          }}
+        />
+      </EuiFormRow>
+
+      {currentFormat ? (
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.indexPattern.decimalPlacesLabel', {
+            defaultMessage: 'Decimals',
+          })}
+          display="rowCompressed"
+        >
+          <EuiFieldNumber
+            data-test-subj="indexPattern-dimension-formatDecimals"
+            value={typeof currentDecimals === 'number' ? currentDecimals : 3}
+            min={0}
+            max={20}
+            onChange={e => {
+              onChange({
+                id: (selectedColumn.params as { format: { id: string } })?.format!.id,
+                params: {
+                  decimals: Number(e.target.value),
```

Comment:
```suggestion
                  decimals: e.target.valueAsNumber,
```

Should work too?

Reply:
Yes you're right. I plainly did some check if the API is available in IE11 before posting that, but it seems even despite it's available IE11 still has a couple of issues around it. Checking the repo shows we're actually still using that API in several other places, so I assume IE11 anyway might have some issues there :man_shrugging: